openedx_learning/core/components/api.py

```diff
@@ -73,15 +73,98 @@ def create_component_version(
     return component_version
 
 
+def create_next_version(
+    component_pk: int,
+    title: str,
+    content_to_replace: dict[str, int | None],
```

Comment:
```suggestion
    content_to_replace: dict[str, int],
    content_to_remove: set[str],
```
non-blocking--

`None` is acting a sentinel value in `content_to_replace`, and I tend to be wary of sentinels, especially when they feel like special cases both on the caller's side and the implementer's side. In this case, I think I'd split into a dict for replacements and a set for deletions.

That said, I think you have a better idea of the calling patterns here, so maybe this will be more elegant to callers than I'm imagining. Regardless, it's a minor thing, and you've explained it very clearly in the docstring, so no sweat if you decide to keep it as-is.

Reply:
I think the big thing that using `None` has going for it is that it makes it harder to call with inconsistent arguments. Say someone passed in:

```python
content_to_replace = {'block.xml': 201, 'hello.png': 928}
content_to_delete = {'hello.png'}
```

If those are both sent in the same call, we have to either let one or the other directive take precedence, or raise an error with the inconsistency, and callers have to be aware of that. In the current implementation, the semantics of a dict make it so that you can't represent that particular inconsistency.

Reply:
I hadn't thought of that. Sounds good to me!